db/patch-group-table-change-weight-column-to-int-.sql

```diff
@@ -0,0 +1,42 @@
+-- change column weight from boolean to integer in group and group_monitor tables
+PRAGMA foreign_keys=off;
+BEGIN TRANSACTION;
+
+CREATE TABLE IF NOT EXISTS [new_group]
+(
+    id           INTEGER      not null
+        constraint group_pk
+            primary key autoincrement,
+    name         VARCHAR(255) not null,
+    created_date DATETIME              default (DATETIME('now')) not null,
+    public       BOOLEAN               default 0 not null,
+    active       BOOLEAN               default 1 not null,
+    weight       INTEGER      NOT NULL DEFAULT 1000,
+    status_page_id INTEGER
+);
+
+CREATE TABLE [new_monitor_group]
+(
+    [id]         INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+    [monitor_id] INTEGER                           NOT NULL REFERENCES [monitor] ([id]) ON DELETE CASCADE ON UPDATE CASCADE,
+    [group_id]   INTEGER                           NOT NULL REFERENCES [group] ([id]) ON DELETE CASCADE ON UPDATE CASCADE,
```

Comment:
Wait, shouldn't the key join the new table?

Actually, I'm not sure if this will link the table to the old `group` table that will be removed.

Reply:
Well, I'd say from the tests that it links to new table since "foreign_keys=on" is only at the end of the script.

Reply:
Possible. I'm not an expert on SQLite syntax 😁